### PR TITLE
feat(aggregations): display aggregation results

### PR DESCRIPTION
**Pull request by mabaasit**
feat(aggregations): display aggregation results COMPASS-5669

1. Resolves open issues in #2930 and created a follow up [COMPASS-5681](https://jira.mongodb.org/browse/COMPASS-5681) for error state.
2. Updated CancelLoader in compass components.

https://user-images.githubusercontent.com/1305718/162007193-78e87541-609d-4bf3-a24a-cfc5612ef3be.mov

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/pipeline-name.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export const PipelineName: React.FunctionComponent<PipelineNameProps> = ({
           <Body>{name}</Body>
         </Tooltip>
       )}
-      {isModified && <span className={modifiedStyles}> - modified</span>}
+      {isModified && <span className={modifiedStyles}>&nbsp;– modified</span>}
```
Comment (alenakhineika) [inline packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/pipeline-name.tsx:48]: Is it intentional to add &nbsp; only on the left side?

**Review by gribnoysup — COMMENTED** [review]
When in "Running" state, maybe it's worth not hiding pagination completely to avoid content flashing in and out all the time (we handle it similarly in the CRUD view), what do you think?

File: packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/pipeline-name.tsx (hunk 1 above)
Comment (mabaasit) [reply packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-settings/pipeline-name.tsx:48]: Yeah! need an empty space between Title and modified flag (`pipeline - modified`). And while rendering, the white-space is not shown.

**Comment by mabaasit** [comment]
> When in "Running" state, maybe it's worth not hiding pagination completely to avoid content flashing in and out all the time (we handle it similarly in the CRUD view), what do you think?

Good idea. Just added it.

**Review by addaleax — APPROVED** [review]

File: packages/compass-components/src/components/cancel-loader.tsx [hunk 2]
```diff
@@ -64,16 +30,12 @@ function CancelLoader({
...
-        <i className={spinnerStyle} />
-        {progressText}
-      </div>
-      <div>
-        <button className={buttonStyle} onClick={onCancel}>
-          {cancelText}
-        </button>
-      </div>
+    <div className={containerStyles} data-testid={dataTestId}>
+      <SpinLoader size="24px" />
```
Comment (mcasimir) [inline packages/compass-components/src/components/cancel-loader.tsx:34]: `spacing[4]`?

File: packages/compass-crud/src/components/document-list.less [hunk 3]
```diff
@@ -154,3 +154,9 @@
     }
   }
 }
+
+.loader {
```
Comment (mcasimir) [inline packages/compass-crud/src/components/document-list.less:158]: just curious, why was this necessary?

File: packages/compass-schema/src/components/compass-schema/compass-schema.module.less [hunk 4]
```diff
@@ -23,6 +23,12 @@
   flex-shrink: 1;
 }
 
+.loader {
```
Comment (mcasimir) [inline packages/compass-schema/src/components/compass-schema/compass-schema.module.less:26]: mostly the same question as above :)

**Review by mcasimir — APPROVED** [review]

**Comment by mcasimir** [comment]
🔥🔥🔥

File: packages/compass-crud/src/components/document-list.less (hunk 3 above)
Comment (mabaasit) [reply packages/compass-crud/src/components/document-list.less:158]: nice. the `CancelLoader` component previously centered the content and also added `height: 100%`.  I removed that and made component occupy width/height it needed and let consumer of the component style it as needed.

**Review by gribnoysup — APPROVED** [review]
Seems like disabling stages still shows them in the pipeline preview:

image

I wonder if there is something we can do to make it more obvious that some stages are disabled. I'll add something for this to the design sync.

Another small thing is that scrollbar here is stuck to the items, overlapping with them:

image

whereas on the builder side it is at the side of the screen. Would be nice to align them, maybe as a follow-up PR.

Otherwise looks really good and works perfectly, great work!
